### PR TITLE
Mark api blocks customer store cms user directory

### LINKED ISSUE #32048: Mark Customer, User, Directory, Store and Cms Blocks as API
The following Blocks should be marked as API in 2.5-develop branch

1. Magento\Customer\Block\SectionConfig
1. Magento\Customer\Block\CustomerData
1. Magento\Customer\Block\CustomerScopeData
1. Magento\Customer\Block\Newsletter
1. Magento\Customer\Block\Address\Grid
1. Magento\Customer\Block\Address\Book
1. Magento\Customer\Block\Address\Edit
1. Magento\Customer\Block\Form\Edit
1. Magento\Customer\Block\Form\Register
1. Magento\Customer\Block\Form\Login
1. Magento\Customer\Block\Form\Login\Info
1. Magento\Customer\Block\Account\SortLinkInterface
1. Magento\Customer\Block\Account\Resetpassword
1. Magento\Customer\Block\Account\Delimiter
1. Magento\Customer\Block\Account\Customer
1. Magento\Customer\Block\Account\Dashboard
1. Magento\Customer\Block\Account\Link
1. Magento\Customer\Block\Account\AuthorizationLink
1. Magento\Customer\Block\Account\Forgotpassword
1. Magento\Customer\Block\Account\AuthenticationPopup
1. Magento\Customer\Block\Account\Dashboard\Address
1. Magento\Customer\Block\Account\Dashboard\Info
1. Magento\User\Block\Buttons
1. Magento\Directory\Block\Currency
1. Magento\Store\Block\Switcher
1. Magento\Cms\Block\Page
1. Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Content
1. Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Tree
1. Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Content\Files

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Mark blocks in modules customer, cms, store, user, directory as API

marked api annotation will keep remain as before only update following list in issue link below
All most class already marked api. We only need add strict_type

Magento\Customer\Block\SectionConfig
Magento\Customer\Block\CustomerData
Magento\Customer\Block\CustomerScopeData
Magento\Customer\Block\Newsletter
Magento\Customer\Block\Address\Grid (already marked api)
Magento\Customer\Block\Address\Book
Magento\Customer\Block\Address\Edit
Magento\Customer\Block\Form\Edit
Magento\Customer\Block\Form\Register
Magento\Customer\Block\Form\Login
Magento\Customer\Block\Form\Login\Info
Magento\Customer\Block\Account\SortLinkInterface
Magento\Customer\Block\Account\Resetpassword
Magento\Customer\Block\Account\Delimiter
Magento\Customer\Block\Account\Customer
Magento\Customer\Block\Account\Dashboard
Magento\Customer\Block\Account\Link
Magento\Customer\Block\Account\AuthorizationLink
Magento\Customer\Block\Account\Forgotpassword
Magento\Customer\Block\Account\AuthenticationPopup
Magento\Customer\Block\Account\Dashboard\Address
Magento\Customer\Block\Account\Dashboard\Info
Magento\User\Block\Buttons
Magento\Directory\Block\Currency
Magento\Store\Block\Switcher
Magento\Cms\Block\Page
Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Content  (already marked api)
Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Tree
Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Content\Files

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32048

### Manual testing scenarios (*)

1. ...
2. ...

### Questions or comments

CC @sivaschenko re-check @api mark and cleanup

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Rebased 2.5-develop

**Comment by mrtuvn** [comment]
@magento run all tests

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @mrtuvn could you please remove the `@author` annotation from the following classes?

- Magento\Customer\Block\Address\Book
- Magento\Customer\Block\Form\Login

Thank you.

**Comment by mrtuvn** [comment]
Of course let's see it again

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Magento Health Index, Magento Health Index, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Integration Tests, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Integration Tests, Functional Tests B2B

**Comment by eduard13** [comment]
@mrtuvn please review the failed Integration Tests.

**Comment by mrtuvn** [comment]
I'm not sure about this! Maybe regression internal bug. Magento try to run strlen something but in that time that first argument not available or not initial so we got error in test

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Unit Tests, Static Tests

**Comment by eduard13** [comment]
@magento run Integration Tests

File: app/code/Magento/Customer/Block/Account/Delimiter.php [hunk 1]
```diff
@@ -15,8 +16,7 @@
 class Delimiter extends \Magento\Framework\View\Element\Template implements SortLinkInterface
 {
     /**
-     * {@inheritdoc}
-     * @since 101.0.0
```
Comment (eduard13) [inline app/code/Magento/Customer/Block/Account/Delimiter.php:19]: Why `@since` annotation has been removed? Please revert it.

Comment (mrtuvn) [reply app/code/Magento/Customer/Block/Account/Delimiter.php:19]: corrected that

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B, Functional Tests EE, Integration Tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests, WebAPI Tests, Functional Tests EE

**Comment by eduard13** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Review by eduard13 — APPROVED** [review]

**Comment by mrtuvn** [comment]
Retarget branch to 2.4-develop

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Integration Tests, Unit Tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by engcom-Oscar** [comment]
Hi @mrtuvn !
It looks like 2 files are missing here 
Magento\Customer\Block\Address\Grid
Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Content
May you please recheck?

**Comment by mrtuvn** [comment]
@engcom-Oscar 
Block 
Magento\Customer\Block\Address\Grid
Magento\Cms\Block\Adminhtml\Wysiwyg\Images\Content

Already marked as api

**Comment by mrtuvn** [comment]
@magento Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by gabrieldagama** [comment]
Since there are not API markings on this PR and some code changes, moved it to 2.5 part of the product.

**Comment by mrtuvn** [comment]
Noted i will change it later

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@gabrieldagama can i close this now ?

**Pull request closed without merging**